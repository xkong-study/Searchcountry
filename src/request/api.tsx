import request from './request'

export const SearchApi = (params:any) => request.get('/country', params)
export const LngLatApi = (params:any) => request.get('',params)
