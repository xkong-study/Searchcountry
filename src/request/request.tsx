import axios from 'axios'

const axiosOption = {
    baseURL: '/',
    timeout: 1000
}

// @ts-ignore
const instance = axios.create(axiosOption);

instance.interceptors.request.use(
    (config) => config,
    async (error) => Promise.reject(error)
);

instance.interceptors.response.use(
    (response) => response.data,
    async (error) => Promise.reject(error)
);

export default instance;
