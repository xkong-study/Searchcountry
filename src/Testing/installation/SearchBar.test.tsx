import React from 'react';
import { shallow } from 'enzyme';
import Autocomplete from '@mui/material/Autocomplete'; // 替换为你的Autocomplete组件的导入路径
import TextField from '@mui/material/TextField'; // 替换为你的TextField组件的导入路径
import Enzyme from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import { SearchApi } from '../../request/api';

Enzyme.configure({ adapter: new Adapter() });


describe('Autocomplete component', () => {
    let options = ['Ireland', 'France', 'Germany']; // replace with your option list

    it('should render Autocomplete component correctly', () => {
        const wrapper = shallow(
            <Autocomplete
                includeInputInList
                options={options}
                renderInput={(params) => <TextField {...params} label="ENTER COUNTRY" variant="standard" />}
            />
        );

        expect(wrapper).toMatchSnapshot(); // Use snapshot tests to ensure components render correctly
    });

});

jest.mock('../../request/api', () => ({
    SearchApi: jest.fn(() => ([{
        common_name: 'Ireland',
        official_name: 'Republic of Ireland',
        short_name: 'IE',
        capital: ["Dublin"],
        currency: {"EUR": {"name": "Euro", "symbol": "€"}},
        flag: "https://flagcdn.com/w320/ie.png",
        region: "Europe",
        timezone: ["UTC"],
        lng: -8,
        lat: 53
    }]))
}));

test('mock search country api test', () => {
    // Call the simulated SearchApi function
    const result = SearchApi({ params: { name: 'Ireland' } });

    // Assert that the SearchApi function is called
    expect(SearchApi).toHaveBeenCalledTimes(1);

    // Assert the parameters of the SearchApi function
    expect(SearchApi).toHaveBeenCalledWith({ params: { name: 'Ireland' } });

    expect(result).toMatchSnapshot();
});
