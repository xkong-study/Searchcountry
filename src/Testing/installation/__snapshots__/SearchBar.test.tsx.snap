// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete component should render Autocomplete component correctly 1`] = `ShallowWrapper {}`;

exports[`mock search country api test 1`] = `undefined`;
